### PR TITLE
Add flux deployment workaround

### PR DESCRIPTION
**Pull request by bernardjkim**
Similarly to https://github.com/gravitational/teleport/pull/40782, this PR documents a temporary workaround to deploy Teleport using FluxCD.

### CODE REVIEW

**Review by hugoShaka — APPROVED** [review]

File: docs/pages/upgrading/cloud-kubernetes.mdx [hunk 1]
```diff
@@ -124,23 +124,26 @@ To suspend automatic updates for an agent, annotate the agent deployment with
...
 
-## ArgoCD deployments
+## GitOps
 
-Automatic updates is not currently compatible with ArgoCD deployments. The Helm chart
-owns the version of the `teleport-agent` resource, so when the `teleport-agent-updater`
-modifies the image version of the `teleport-agent` resource, ArgoCD reports the `teleport-agent`
-resource as `OutOfSync`.
+Automatic updates is incompatible with some GitOps tools used for continuous deployment.
+The Teleport Helm chart owns the version of the `teleport-agent` resource, so when
```
Comment (ptgott) [inline docs/pages/upgrading/cloud-kubernetes.mdx:130]: Which Helm chart?

**Review by ptgott — APPROVED** [review]

**Comment by ptgott** [comment]
Are we backporting this one?

**Comment by fheinecke** [comment]
IMO we should probably add tests for this (and the ArgoCD one as well) if we're advertising this change to our customers.

Let me know if we want to do this - internal tools team has Terraform modules to easily spin up new k8s clusters managed by Flux.

**Comment by bernardjkim** [comment]
> IMO we should probably add tests for this (and the ArgoCD one as well) if we're advertising this change to our customers.

Yeah, that would be a great idea. Where would be a good place to look to get started writing tests for Flux/Argo?

**Comment by fheinecke** [comment]
> Where would be a good place to look to get started writing tests for Flux/Argo?

@bernardjkim we should probably chat next week about this.

**Pull request merged**